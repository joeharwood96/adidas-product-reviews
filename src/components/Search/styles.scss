.search {

    &-wrapper {
        display: flex;
        min-width: 140px;
        max-width: 300px;
        border: solid thin #e9ecef;

        &:hover {
            transition: border ease 1s;
        }

        &:focus-within {
            border: solid thin #000000;
        }
    }

    &-input {
        padding-left: 10px;
        padding-right: 35px;
        background-color: #e9ecef;
        text-overflow: ellipsis;
        border: none;
        outline: none;
    
    }

    &-button {
        background-color: #e9ecef;
        border: solid thin #e9ecef;
        
        &:hover {
            cursor: pointer;
        }
    }
}