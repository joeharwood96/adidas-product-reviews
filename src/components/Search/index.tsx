import React, { useState } from 'react'
import './styles.scss'
import { HiSearch } from 'react-icons/hi'
import { SearchContext } from '../../App'

export default function Search() {

    return (
        <SearchContext.Consumer>
            {
                searchContext => {
                   return <div className="search-wrapper">
                        <input 
                            type="text" 
                            className="search-input" 
                            placeholder="Search"
                            value={searchContext?.searchString}
                            onChange={searchContext?.onChange}
                        />
                        <button className="search-button">
                            <HiSearch size="2rem" className="search-icon"/>
                        </button>
                    </div>
                }
            }
        </SearchContext.Consumer>
    )
}