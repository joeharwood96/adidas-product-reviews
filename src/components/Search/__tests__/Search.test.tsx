import React from "react";
import {fireEvent, render} from '@testing-library/react';  
import Search from "../index";

describe("<Search />", () => {
    it("should display the search", () => {
        const component = <Search />
        expect(component).toMatchSnapshot()
    })

    it("updates on change", () => {
        const { queryByPlaceholderText } = render(<Search />)
        const searchInput = queryByPlaceholderText("Search")
        
        fireEvent.change(searchInput as HTMLInputElement, {target: { value: "test"}})
        expect((searchInput as HTMLInputElement).value).toEqual("test")
    })
});