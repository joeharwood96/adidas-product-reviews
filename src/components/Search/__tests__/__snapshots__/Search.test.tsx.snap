// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Search /> should display the search 1`] = `<Search />`;
